lib/core/src/mqtt.rs

```diff
@@ -0,0 +1,122 @@
+use defmt::*;
+#[cfg(not(feature = "std"))]
+use heapless::String;
+use rust_mqtt::{
+    client::{client::MqttClient, client_config::ClientConfig},
+    packet::v5::reason_codes::ReasonCode,
+    utils::rng_generator::CountingRng,
+};
+use serde::{Deserialize, Serialize};
+
+#[cfg(not(feature = "std"))]
+pub struct MqttMessage {
+    pub topic: String<48>,
+    pub payload: String<512>,
+}
+
+#[cfg(feature = "std")]
+pub struct MqttMessage {
+    pub topic: String,
+    pub payload: String,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct ButtonMqttMessage {
+    pub task_id: u8,
+    pub status: bool,
+}
+
+pub struct HypedMqttClient<
+    'a,
+    T: embedded_io_async::Read + embedded_io_async::Write,
+    R: rand_core::RngCore,
+> {
+    pub client: MqttClient<'a, T, 5, R>,
+}
+
+pub fn initialise_mqtt_config() -> ClientConfig<'static, 5, CountingRng> {
+    let mut config = ClientConfig::new(
+        rust_mqtt::client::client_config::MqttVersion::MQTTv5,
+        CountingRng(20000),
+    );
+    config.add_max_subscribe_qos(rust_mqtt::packet::v5::publish_packet::QualityOfService::QoS1);
```

Comment:
I can't remember if you said this supports QoS2 or not?

Reply:
Only supports QoS 1 currently, i was going to modify a fork of the library at some point to add QoS2

Reply:
(We've just had a discussion about this on Slack, QoS2 not needed in favour of manual acks.)